GitCommands/Git/GitCommandHelpers.cs

```diff
@@ -58,45 +58,53 @@ public static class GitCommandHelpers
     {
         private static readonly ISshPathLocator SshPathLocatorInstance = new SshPathLocator();
 
-        public static void SetEnvironmentVariable(bool reload = false)
+        #region Environment variables
+
+        /// <summary>
+        /// The <c>USER</c> environment variable's value for the user/machine.
+        /// </summary>
+        [CanBeNull]
+        private static readonly string UserHomeDir
+            = Environment.GetEnvironmentVariable("HOME", EnvironmentVariableTarget.User)
+           ?? Environment.GetEnvironmentVariable("HOME", EnvironmentVariableTarget.Machine);
+
+        /// <summary>
+        /// Sets <c>PATH</c>, <c>HOME</c>, <c>TERM</c> and <c>SSH_ASKPASS</c> environment variables
+        /// for the current process.
+        /// </summary>
+        public static void SetEnvironmentVariables()
         {
-            string path = Environment.GetEnvironmentVariable("PATH", EnvironmentVariableTarget.Process);
-            if (!string.IsNullOrEmpty(AppSettings.GitBinDir) && !path.Contains(AppSettings.GitBinDir))
-            {
-                Environment.SetEnvironmentVariable("PATH", string.Concat(path, ";", AppSettings.GitBinDir), EnvironmentVariableTarget.Process);
-            }
+            // PATH variable
 
-            if (!string.IsNullOrEmpty(AppSettings.CustomHomeDir))
+            if (!string.IsNullOrEmpty(AppSettings.GitBinDir))
             {
-                Environment.SetEnvironmentVariable(
-                    "HOME",
-                    AppSettings.CustomHomeDir);
-                return;
-            }
+                // Ensure the git binary dir is on the path
+                string path = Environment.GetEnvironmentVariable("PATH");
 
-            if (AppSettings.UserProfileHomeDir)
-            {
-                Environment.SetEnvironmentVariable(
-                    "HOME",
-                    Environment.GetEnvironmentVariable("USERPROFILE"));
-                return;
+                if (path == null)
+                {
+                    Environment.SetEnvironmentVariable("PATH", AppSettings.GitBinDir);
+                }
+                else if (!path.Contains(AppSettings.GitBinDir))
+                {
+                    Environment.SetEnvironmentVariable("PATH", $"{path};{AppSettings.GitBinDir}");
```

Comment:
I know that mono support is dropped but just in case we can use
https://msdn.microsoft.com/en-us/library/system.io.path.pathseparator(v=vs.110).aspx

Reply:
Good idea. Done.